stock_valuation_fifo_lot/models/stock_move.py

```
@@ -0,0 +1,97 @@
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            move = move.with_context(used_in_move_id=move.id)
+            layer = super(StockMove, move)._create_out_svl(
+                forced_quantity=forced_quantity
+            )
+            layers |= layer
+        return layers
+
+    def _create_in_svl(self, forced_quantity=None):
+        """
+        1. Check stock move - Multiple lot on the stock move is not
+           allowed for incoming transfer
+        2. Change product standard price to first available lot price
+        """
+        layers = self.env["stock.valuation.layer"]
+        for move in self:
+            layer = super(StockMove, move)._create_in_svl(
+                forced_quantity=forced_quantity
+            )
+            # Calculate standard price (Sorted by lot created date)
+            if (
+                move.product_id.cost_method == "fifo"
+                and move.product_id.tracking != "none"
+            ):
+                all_candidates = move.product_id.with_context(
+                    sort_by="lot_create_date"
+                )._get_fifo_candidates(move.company_id)
+                if all_candidates:
+                    move.product_id.sudo().with_company(
+                        move.company_id.id
+                    ).with_context(
+                        disable_auto_svl=True
+                    ).standard_price = all_candidates[
+                        0
+                    ].unit_cost
+            layers |= layer
+        return layers
+
+    def _get_price_unit(self):
+        """
+        No PO, Get price unit from lot price
+        """
+        self.ensure_one()
+        if not self.company_id.use_lot_get_price_unit_fifo:
+            return super()._get_price_unit()
+        if (
+            not self.purchase_line_id
+            and self.product_id.cost_method == "fifo"
+            and len(self.lot_ids) == 1
```

Should we raise an error when `len(self.lot_ids) > 1`?

---

Do you have any concerns or inconveniences if we use Odoo's standard behavior for this case?

---

> Should we raise an error when `len(self.lot_ids) > 1`?

I think we could make it configurable if the community requires it. I don't have a strong need for that at the moment.

---

Okay, single lot per SVL would not work when migrating/installing with existing data.  So, I don't think we can do that.